src/main/java/picard/sam/util/SamComparison.java

```diff
@@ -397,6 +416,10 @@ private boolean alignmentsMatch(final SAMRecord s1, final SAMRecord s2) {
                 (samComparisonArgumentCollection.LENIENT_UNKNOWN_MQ_ALIGNMENT && s1.getMappingQuality() == SAMRecord.UNKNOWN_MAPPING_QUALITY && s2.getMappingQuality() == SAMRecord.UNKNOWN_MAPPING_QUALITY));
     }
 
+    private void compareAndUpdateMappingQualityConcordance(final SAMRecord s1, final SAMRecord s2) {
+        mappingQualityHistogram.increment(new Pair<>(s1.getMappingQuality(), s2.getMappingQuality()).toString());
```

Comment:
I don't think creating a pair object and using its toString method is the best way to  generate this string.  Just build the string yourself instead.  You can write a one line method for it for clarity.